addons/spreadsheet/static/src/data_sources/data_source.js

```diff
@@ -79,6 +80,18 @@ export class LoadableDataSource {
         return this._isFullyLoaded;
     }
 
+    /**
+     * @returns {import("@odoo/o-spreadsheet").FPayload | undefined}
+     */
+    getLoadingError() {
```

Comment:
I'm not a big fan of `getLoadingError`, It's not natural from the pov of the function to get the loading error. 

```js
/**
 * @protected
 */
_assertDataIsLoaded(throwError=true) {
    if (!this._isFullyLoaded) {
        this.load();
        if (throwError) throw LOADING_ERROR;
        else return LOADING_ERROR;
    }
    if (!this._isValid) {
        if (throwError) throw ...
        else return ...
    }
}

What do you think to reuse assertDataIsLoaded (made public) with a parameter ? It will be easier to read IMO
```